test-data/unit/check-generics.test

```diff
@@ -851,3 +851,9 @@ def f3(x: A, y: B) -> B: ...
 g = f1
 g = f2
 g = f3
+
+[case testTypeVariableWithContainerAndTuple]
+from typing import TypeVar, Container
+T = TypeVar('T')
+def f(x: Container[T]) -> T: ...
+f((1, 2))
```

Comment:
Can you add `reveal_type()` around this expression and update the output with the expected error message that generates? (It's not really an error but it is formatted like an error.)

Otherwise looks good!